### ISSUE TITLE
Monitor detected with active workspace but blank screen on Hyprland Archlinux with Nvidia/intel laptop

### ISSUE THREAD

**Issue by ghost** [hyprwm/Hyprland#6343]
### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit fefa55d406e38eda481ce75178cef3e43298762d  (build: fix non-pch build (6337)).
Date: Wed Jun 5 04:42:44 2024
Tag: v0.40.0-157-gfefa55d4, commits: 4763

flags: (if any)


System Information:
System name: Linux
Node name: arch
Release: 6.9.3-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Fri, 31 May 2024 15:14:45 +0000


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation CoffeeLake-H GT2 [UHD Graphics 630] [8086:3e9b] (prog-if 00 [VGA controller])
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GP107M [GeForce GTX 1050 Ti Mobile] [10de:1c8c] (rev a1) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

======Config-Start======


Config File: /home/user/.config/hypr/hyprland.conf: Read Succeeded
# This is an example Hyprland config file.
#
# Refer to the wiki for more information.

#
# Please note not all available settings / options are set here.
# For a full list, see the wiki
#

# See https://wiki.hyprland.org/Configuring/Monitors/
monitor=eDP-1,1920x1080@60,2560x0,1
monitor=HDMI-A-1,1920x1080@59.95,0x0,1

# See https://wiki.hyprland.org/Configuring/Keywords/ for more
debug:disable_logs = false
#env = WLR_DRM_DEVICES,$HOME/.config/hypr/card
#env = _JAVA_AWT_WM_NONREPARENTING,1
#env = XCURSOR_SIZE,24

#env = LIBVA_DRIVER_NAME,nvidia
#env = XDG_SESSION_TYPE,wayland
#env = GBM_BACKEND,nvidia-drm
#env = __GLX_VENDOR_LIBRARY_NAME,nvidia

# Execute your favorite apps at launch
# exec-once = waybar & hyprpaper & firefox

# Source a file (multi-file configs)
# source = ~/.config/hypr/myColors.conf

# Set programs that you use

# Some default env vars.
env = XCURSOR_SIZE,24
#env = LIBVA_DRIVER_NAME,nvidia
#env = XDG_SESSION_TYPE,wayland
#env = GBM_BACKEND,nvidia-drm
#env = __GLX_VENDOR_LIBRARY_NAME,nvidia

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/


#decoration {
#    # See https://wiki.hyprland.org/Configuring/Variables/ for more
#
#    rounding = 5
#
#    blur {
#        enabled = true
#        size = 3
#        passes = 1
#        
#        vibrancy = 0.15
#    }
#
#    drop_shadow = true
#    shadow_range = 4
#    shadow_render_power = 3
#    col.shadow = rgba(1a1a1aee)
#}

dwindle {
preserve_split = true
}

animations {

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 7, myBezier
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default
}



# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more


# See https://wiki.hyprland.org/Configuring/Keywords/ for more
$mainMod = SUPER

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
bind = $mainMod, Q, exec, foot
bind = $mainMod, C, killactive,
bind = $mainMod, M, exit,
#bind = $mainMod, E, exec, $fileManager
bind = $mainMod, V, togglefloating,
bind = $mainMod, P, exec, fuzzel
bind = $mainMod, R, pseudo, # dwindle
bind = $mainMod, S, togglesplit, # dwindle

# Move focus with mainMod + arrow keys
bind = $mainMod, H, movefocus, l
bind = $mainMod, L, movefocus, r
bind = $mainMod, K, movefocus, u
bind = $mainMod, J, movefocus, d

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10

# Example special workspace (scratchpad)
bind = $mainMod, S, togglespecialworkspace, magic
bind = $mainMod SHIFT, S, movetoworkspace, special:magic

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow
bindm = SUPER_CTRL, mouse:272, resizewindow

======Config-End========


```

</details>

### Bug or Regression?

Bug

### Description

Monitor: Samsung Odyssey G50A
GPU: NVidia GTX 1050TI
CPU/dGPU: Intel CoffeeLake-H GT2 [UHD Graphics 630]
laptop: Acer Nitro 5

kernels linux-zen and linux give same result with nvidia and nvidia-dkms(nvidia-open didn't work with gpu)
I'm using the following in config, yet get blank monitor with active workspace(SUPER+2 makes mouse disappear, whereas this doesn't happen when gpu isn't recognized in running hyprland) no matter the following as long as the above is in place
```
env = LIBVA_DRIVER_NAME,nvidia
env = XDG_SESSION_TYPE,wayland
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
```
not sure what's wrong, I know nvidia is working because the kernels and drivers i mentioned are he only ones that give any nvidia-smi output after running nvx(nvidia-exec)

this all occured after an update after not updating for like 1.5 weeks; I'm not sure what changed it since I've tried reverting nvidia and linux drivers back to 2023 releases that worked fine for me and I can't recall what changed, despite the monitor working fine with this setup for the past year. I might be able to provide a list of changes from that time period from /var/log/pacman.log but it would have like 6 GiB worth of packages in it to sift through so I'll only post it if necessary/requested by someone who is legitimately able to help.

tl;dr: why no monitor worky?!

### How to reproduce

install and attempt to run above setup

**Comment by User10987654321** [comment]
posting the following bc might provide more useful info (i.e. currentFormat in eDP-1 is XRGB888 and HDMI-A-1 is invalid)

 hyprctl monitors
```Monitor eDP-1 (ID 0):
1920x1080@60.05200 at 2560x0
description: AU Optronics 0x61ED
make: AU Optronics
model: 0x61ED
serial:
active workspace: 1 (1)
special workspace: 0 ()
reserved: 0 0 0 0
scale: 1.00
transform: 0
focused: yes
dpmsStatus: 1
vrr: 0
activelyTearing: false
disabled: false
currentFormat: XRGB8888
availableModes: 1920x1080@60.05Hz

Monitor HDMI-A-1 (ID 1):
2560x1440@144.00000 at 0x0
description: Samsung Electric Company Odyssey G50A HCPTB00121
make: Samsung Electric Company
model: Odyssey G50A
serial: HCPTB00121
active workspace: 2 (2)
special workspace: 0 ()
reserved: 0 0 0 0
scale: 1.00
transform: 0
focused: no
dpmsStatus: 1
vrr: 0
activelyTearing: false
disabled: false
currentFormat: Invalid
availableModes: 2560x1440@144.00Hz 2560x1440@120.00Hz 2560x1440@59.95Hz 1920x1080@143.98Hz 1920x1080@119.88Hz 1920x1080@59.94Hz 1680x1050@59.95Hz 1600x900@60.00Hz 1280x1024@75.03Hz 1280x1024@60.02Hz 1440x900@59.89Hz 1280x800@59.81Hz 1152x864@75.00Hz 1280x720@60.00Hz 1280x720@59.94Hz 1024x768@75.03Hz 1024x768@70.07Hz 1024x768@60.00Hz 800x600@75.00Hz 800x600@72.19Hz 800x600@60.32Hz 800x600@56.25Hz 720x480@59.94Hz 640x480@75.00Hz 640x480@72.81Hz 640x480@59.94Hz
```

**Comment by igorgue** [comment]
OH... same just happened to me, I tried `nvx` then removing it made a mess...

**Comment by igorgue** [comment]
Looks like format is the GBM backend and I have it properly set. 

Another thing I noticed is that the 2 cards I have flipped at boot and nvidia is 1 and amd is 0...

On another note, Gnome Wayland is able to output the monitor...

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
